config/checkstyle/checkstyle.xml

```diff
@@ -0,0 +1,120 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
+        "https://checkstyle.org/dtds/configuration_1_3.dtd">
+
+<module name="Checker">
+  <property name="charset" value="UTF-8" />
+
+  <module name="SuppressionFilter">
+    <property name="file" value="${suppressions}" />
+  </module>
+
+  <!-- Checks Java files and forbids empty Javadoc comments -->
+  <module name="RegexpMultiline">
+    <property name="id" value="EmptyJavadoc"/>
+    <property name="format" value="\/\*[\s\*]*\*\/"/>
+    <property name="fileExtensions" value="java"/>
+    <property name="message" value="Empty javadoc comments are forbidden"/>
+  </module>
+
+  <!-- 
+    We include snippets that are wrapped in `// tag` and `// end` into the
+    docs, stripping the leading spaces. If the context is wider than 76
+    characters then it'll need to scroll. This fails the build if it sees
+    such snippets.
+  -->
+  <module name="RegexpMultiline">
+    <property name="id" value="SnippetLength"/>
+    <property name="format" value="^( *)\/\/\s*tag(.+)\s*\n(.*\n)*\1.{77,}\n(.*\n)*\1\/\/\s*end\2\s*$"/>
+    <property name="fileExtensions" value="java"/>
+    <property name="message" value="Code snippets longer than 76 characters get cut off when rendered in the docs"/>
+  </module>
+
+  <!-- Checks that no tab characters are used in source files -->
+  <module name="FileTabCharacter">
+    <property name="eachLine" value="true"/>
+  </module>
+
+  <module name="TreeWalker">
+    <module name="LineLength">
+      <property name="max" value="120"/>
+      <property name="ignorePattern" value="^ *\* *https?://[^ ]+$|^ ?\* +\+\-+|^ ?\* +\||^ ?\* ?\-+"/>
```

Comment:
Does this pattern already ignore `import` or not? Should we ignore it too because some package like our own `opendistroforelasticsearch` is quite long? https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml#L51

Reply:
From official checkstyle documentation (https://checkstyle.sourceforge.io/config_sizes.html#LineLength):

>Package and import statements (lines matching pattern ^(package|import) .*) are not verified by this check.

Reply:
This pattern is for javadoc comments, mostly the long ones we have for tables (state transitions, options, etc.) in comments, which don't fit in 120 char length.